p2p/net/swarm/dial_worker.go

```diff
@@ -358,7 +358,8 @@ loop:
 			}
 
 			// it must be an error -- add backoff if applicable and dispatch
-			if res.Err != context.Canceled && !w.connected {
+			// ErrDialRefusedBlackHole shouldn't end up here, just a safety check
+			if res.Err != ErrDialRefusedBlackHole && res.Err != context.Canceled && !w.connected {
```

Comment:
Should this go in error log? If we're not noisy about it, we'd miss this bug if we ever introduce it.

Reply:
Added this